packages/grafana-ui/src/components/TimePicker/TimeRangePicker/TimePickerContent.tsx

```diff
@@ -146,35 +149,51 @@ interface FormProps extends Omit<Props, 'history'> {
   historyOptions?: TimeOption[];
 }
 
+interface FullscreenFormProps extends FormProps {
+  isRightSideVisible: boolean;
+}
+
 export const TimePickerContentWithScreenSize: React.FC<PropsWithScreenSize> = props => {
+  const { quickOptions = [], otherOptions = [], isFullscreen, hideQuickRanges, hideOtherRanges } = props;
+  const isRightSideVisible = !isFullscreen || !hideQuickRanges || !hideOtherRanges;
   const theme = useTheme();
-  const styles = getStyles(theme, props.isReversed);
+  const styles = getStyles(theme, props.isReversed, isRightSideVisible);
   const historyOptions = mapToHistoryOptions(props.history, props.timeZone);
-  const { quickOptions = [], otherOptions = [], isFullscreen } = props;
 
   return (
     <div className={cx(styles.container, props.className)}>
       <div className={styles.body}>
         <div className={styles.leftSide}>
-          <FullScreenForm {...props} visible={isFullscreen} historyOptions={historyOptions} />
+          <FullScreenForm
+            {...props}
+            visible={isFullscreen}
+            historyOptions={historyOptions}
+            isRightSideVisible={isRightSideVisible}
+          />
         </div>
         <CustomScrollbar className={styles.rightSide}>
           <NarrowScreenForm {...props} visible={!isFullscreen} historyOptions={historyOptions} />
-          <TimeRangeList
-            title="Relative time ranges"
-            options={quickOptions}
-            onSelect={props.onChange}
-            value={props.value}
-            timeZone={props.timeZone}
-          />
-          <div className={styles.spacing} />
-          <TimeRangeList
-            title="Other quick ranges"
-            options={otherOptions}
-            onSelect={props.onChange}
-            value={props.value}
-            timeZone={props.timeZone}
-          />
+          {!hideQuickRanges && (
+            <>
+              <TimeRangeList
+                title="Relative time ranges"
+                options={quickOptions}
+                onSelect={props.onChange}
+                value={props.value}
+                timeZone={props.timeZone}
+              />
+              <div className={styles.spacing} />
+            </>
+          )}
+          {!hideOtherRanges && (
```

Comment:
I think we could use only one flag here, `hideQuickRanges` and hide both relative ranges with it. I don't think there'll be cases where we'll need only either one of them.

Reply:
Initially I added one single flag but I thought someone might need more granularity on it. I'll switch to one flag only

Reply:
Btw, I saw your other comment regarding `isReversed`, I responded to it but the comment disappeared completely. Should I do that one?

Reply:
Oh, nvm. I found a little tiny edge case that I have to fix. Will be back with an update soon

Reply:
> Btw, I saw your other comment regarding `isReversed`, I responded to it but the comment disappeared completely. Should I do that one?

That was my bad, I thought the reversed prop is being exposed on the picker but it makes sense to have it being exposed on `TimeRangeInput`.

Reply:
Fixed.